src/draw/handler/Draw.Polyline.js

```diff
@@ -278,16 +278,20 @@ L.Draw.Polyline = L.Draw.Feature.extend({
 			if (Math.abs(distance) < 9 * (window.devicePixelRatio || 1)) {
 				this.addVertex(e.latlng);
 			}
+            
+            this._clickHandled = true;            
 		}
 		this._mouseDownOrigin = null;
 	},
 
 	_onTouch: function (e) {
 		// #TODO: use touchstart and touchend vs using click(touch start & end).
-		if (L.Browser.touch) { // #TODO: get rid of this once leaflet fixes their click/touch.
+		if (L.Browser.touch && !this._clickHandled) { // #TODO: get rid of this once leaflet fixes their click/touch.
```

Comment:
Nitpick: If there is an issue in the leaflet repo that relates to this TODO it would be nice if you put the issue number in this comment.